chain/ethereum/src/chain.rs

```diff
@@ -734,6 +741,55 @@ pub struct TriggersAdapter {
     unified_api_version: UnifiedMappingApiVersion,
 }
 
+/// Fetches blocks from the cache based on block numbers, excluding duplicates
+/// (i.e., multiple blocks for the same number), and identifying missing blocks that
+/// need to be fetched via RPC/Firehose. Returns a tuple of the found blocks and the missing block numbers.
+async fn fetch_unique_blocks_from_cache(
```

Comment:
I think this would be a good candidate to add some unit test coverage

Reply:
Added in latest commit